includes/class-sensei-quiz.php

```diff
@@ -1523,6 +1528,53 @@ public static function start_quiz_questions_loop() {
 		$sensei_question_loop['quiz_id']   = $quiz_id;
 	}
 
+	/**
+	 * Display quiz progress bar.
+	 *
+	 * @param int   $quiz_id quiz id.
+	 * @param array $pagination_settings pagination settings.
+	 * @since 3.15.0
+	 */
+	public static function display_progress_bar( $quiz_id, $pagination_settings ) {
+
+		$user_id             = get_current_user_id();
+		$lesson_id           = Sensei()->quiz->get_lesson_id( $quiz_id );
+		$all_questions_count = count( Sensei()->lesson->lesson_quiz_questions( $quiz_id, 'publish' ) );
+		$answers             = Sensei()->quiz->get_user_answers( $lesson_id, $user_id );
+		$answers_count       = is_array( $answers ) ? count(
+			array_filter(
+				$answers,
+				function( $item ) {
+					return is_array( $item );
+				}
+			)
+		) : 0;
+		Sensei()->assets->register( 'sensei-quiz-progress', 'blocks/progress-bar.js' );
+		Sensei()->assets->enqueue(
+			'sensei-shared-blocks-style',
+			'blocks/shared-style.scss',
+			[ 'sensei-shared-blocks-style' ]
+		);
+		wp_enqueue_script( 'sensei-quiz-progress' );
+
+		$radius           = $pagination_settings['progress_bar_radius'];
+		$height           = $pagination_settings['progress_bar_height'];
+		$color            = empty( $pagination_settings['progress_bar_color'] ) ? '' : $pagination_settings['progress_bar_color'];
+		$background_color = empty( $pagination_settings['progress_bar_background'] ) ? '' : $pagination_settings['progress_bar_background'];
+		wp_localize_script(
+			'sensei-quiz-progress',
+			'php_vars',
```

Comment:
Maybe this global variable should be changed to something more specific like `sensei_quiz_progress`?

Reply:
I think it's a good idea to add the `sensei_` prefix. It's a nice practice for global variables. This also makes it consistent with the other `wp_localize_script` calls.

Reply:
https://github.com/Automattic/sensei/pull/4606/files#diff-964a2368a54a4fd7dd7d8dfe593a8f74e690d028590dc23dbb8d1d4e5b68577fR1669